### PR TITLE
DefaultTokenAcquirerFactoryImplementation use ConcurrentDictionary

### PR DESCRIPTION
**Pull request by sammurrayms**
# DefaultTokenAcquirerFactoryImplementation use ConcurrentDictionary

DefaultTokenAcquirerFactoryImplementation now supports multiple regions and more thread safe.

## Description

- The DefaultTokenAcquirerFactoryImplementation now incorporates the region within the key string, resolving a bug wherein getting a token acquirer for a particular region could inadvertently yield a token acquirer meant for a different region. 
- Additionally, the implementation now employs a ConcurrentDictionary to manage its data, enhancing thread safety in scenarios where multiple threads may attempt to create a TokenAcquirer simultaneously.

### CODE REVIEW

**Review by MZOLN — APPROVED** [review]
LGTM

File: src/Microsoft.Identity.Web.TokenAcquisition/DefaultTokenAcquirerFactoryImplementation.cs [hunk 1]
```diff
@@ -112,10 +113,9 @@ private void CheckServiceProviderNotNull()
             }
         }
 
-
-        private static string GetKey(string? authority, string? clientId)
+        private static string GetKey(string? authority, string? clientId, string? region)
         {
-            return $"{authority}{clientId}";
+            return $"{authority}{clientId}{region}";
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.TokenAcquisition/DefaultTokenAcquirerFactoryImplementation.cs:118]: what happens if the region is null?

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @sammurrayms

**Comment by jennyf19** [comment]
You'll need to cherry pick this over to the rel/v2 branch and open a PR. We'll merge it after the next release is out on Tuesday.

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/Microsoft.Identity.Web.TokenAcquisition/DefaultTokenAcquirerFactoryImplementation.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web.TokenAcquisition/DefaultTokenAcquirerFactoryImplementation.cs:118]: @sammurrayms can you check on this one and then we can merge. Thx.

Comment (sammurrayms) [reply src/Microsoft.Identity.Web.TokenAcquisition/DefaultTokenAcquirerFactoryImplementation.cs:118]: Added a test for this case. The resulting string just ignores the region.

```csharp
var no_region = DefaultTokenAcquirerFactoryImplementation.GetKey("1", "2", null);
Assert.Equal("12", no_region);
```

**Comment by sammurrayms** [comment]
https://github.com/AzureAD/microsoft-identity-web/issues/2765

**Pull request merged**